src/AutoHashEqualsCached.jl

```diff
@@ -37,25 +43,27 @@ function _show_default_auto_hash_equals_cached(io::IO, @nospecialize(x))
 end
 
 # Find the first struct declaration buried in the Expr.
-get_struct_decl(typ) = nothing
-function get_struct_decl(typ::Expr)
+get_struct_decl(__source__, typ) = nothing
+function get_struct_decl(__source__, typ::Expr)
```

Comment:
I'm not familiar enough with how Julia macros interact with source code positions. Can you explain a bit what this threading of the `__source__` and the `if x isa LineNumberNode` branches achieve?

Reply:
@nickrobinson251 can you have a peek at this too, since you've recently done macros with nice error reporting?

the more eyes the merrier, i think!

Reply:
Basically the parser intersperses line number nodes into each line of every block.  In a macro, you get a secret parameter __location__ that is the location where the `@` token appeared.  I'm threading it through these calls and tracking the last location seen so that I can place it into any error messages that are produced.